sigridci/sigridci/sigrid_api_client.py

```diff
@@ -104,9 +104,11 @@ def handle401or403(e: urllib.error.HTTPError, msg: str):
 
     @staticmethod
     def printResponse(e: urllib.error.HTTPError):
-        UploadLog.log(f"Response headers:\n{str(e.headers)}")
-        if e.fp is not None:
-            UploadLog.log(f"Response body:\n{e.fp.read().decode()}")
+        UploadLog.log(f"Response headers:\n{dict(e.headers)}")
```

Comment:
If there's no body, we output "No response body". Wouldn't it be better to do the same for headers: if we detect there are 0 headers, we say "No response headers".

This is only error output, so we shouldn't over-engineer, but now it might lead to confusion, with people thinking there actually are headers, but they're somehow invisible (e.g., they start with a terminal control character setting the font color equal to the background).

Reply:
Fair point. I think in a normal situation it's pretty much impossible to have no headers at all. However, we've also had "weird proxy" situations, so in that regard this *could* indeed happen. I'll change it.